Services/PetService.cs

```diff
@@ -8,32 +8,46 @@
 using bootcamp_api.Data;
 using bootcamp_api.Exceptions;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
 
 namespace bootcamp_api.Services
 {
     public class PetService: IPetService
     {
 
         private readonly PawssierContext _context;
+        private readonly IMapper _mapper;
 
-        public PetService(PawssierContext context)
+        public PetService(PawssierContext context, IMapper mapper)
         {
             _context = context;
+            _mapper = mapper;
         }
 
-        public Pet[] GetAll()
+        public Dto.PetListItem[] GetAll(ApiVersion version, int user_id)
         {
             var pets = _context.Pets
                 .Include(p => p.Conditions)
                 .Include(p => p.Vaccines)
                 .Include(p => p.Prescriptions)
                 .Include(p => p.PetPhoto)
-                .Include(p => p.VetRecords);
+                .Include(p => p.VetRecords)
+                .Where(p => p.User_id == user_id);
+
+            var petAry = pets.OrderBy(p => p.Id).ToArray();
+
+            var petListItems = _mapper.Map<Pet[], Dto.PetListItem[]>(petAry);
+
+            foreach(Dto.PetListItem x in petListItems)
+            {
+                x.Link = "/api/v" + version + "/Pets/" + x.Id;
```

Comment:
You will want to generate this link in multiple places such as your Bookmarks, Pet, and PetListItem.

Instead of copying these hard coded string into multiple place you should create a simple helper service.  It can be static class with a static function.

LinkService.GenerateLink(petId)